server/src-exec/Main.hs

```diff
@@ -141,6 +140,16 @@ main =  do
         mkGenericStrLog "event_triggers" "starting workers"
       void $ C.forkIO $ processEventQueue hloggerCtx logEnvHeaders httpSession pool cacheRef eventEngineCtx
 
+      -- start a background thread to check for updates
+      void $ C.forkIO $ checkForUpdates loggerCtx httpManager
+
+      -- start a background thread for telemetry
+      when enableTelemetry $ do
+        unLogger logger $ mkGenericStrLog "telemetry" telemetryNotice
+        res <- getUniqIds ci
```

Comment:
We can merge `initialise` and `migrate` functions into a single function, say `initialise`. This should return the unique ids (database, instance). These unique ids then can be directly used here.